e2etests/node_scan_rhcos_test.go

```diff
@@ -102,6 +103,9 @@ func buildRequest(notes []v1.Note) *v1.GetNodeVulnerabilitiesRequest {
 }
 
 func TestGRPCGetRHCOSNodeVulnerabilities(t *testing.T) {
+	if !features.RHCOSNodeScanning.Enabled() {
```

Comment:
you can set it via `t.Setenv(features.RHCOSNodeScanning.EnvVar())`. Think of it as the same thing as `envisolator` which we used to have, but it resets the env var for us at the end of the test

Reply:
Unfortunately this didn't seem to work: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/stackrox_scanner/1065/pull-ci-stackrox-scanner-master-e2e-tests/1620680953611423744

Reply:
Nope, it does not work. The scanner-under-test for e2e would need to be deployed with that flag enabled and this is not the case currently. See:

```
    Environment:
      POD_NAME:                  scanner-767f474d7-kc8hz (v1:metadata.name)
      ROX_SKIP_PEER_VALIDATION:  true
      ROX_RHCOS_NODE_SCANNING:   false
```

in the `roxci-artifacts/scanner/33227a261a3131cf9921c383a4846feff252d4ac/1620694223667859456-pull-ci-stackrox-scanner-master-e2e-tests/e2e-tests/k8s-service-logs/stackrox/scanner-767f474d7-kc8hz_describe.log` form the [recent CI run ](https://console.cloud.google.com/storage/browser/_details/roxci-artifacts/scanner/33227a261a3131cf9921c383a4846feff252d4ac/1620694223667859456-pull-ci-stackrox-scanner-master-e2e-tests/e2e-tests/k8s-service-logs/stackrox/scanner-767f474d7-kc8hz_describe.log;tab=live_object?authuser=1) (assumes authuser=1 for accessing the logs produced in the CI)

Reply:
I think the `t.Setenv` is good for setting the values in `pkg/env/list.go` (at least that is what the code search suggests) which can be changed at runtime. The vars from `pkg/features/list.go` seem to be built-time variables and maybe they do not allow setting them like that?

Let me try moving the feature flag from `pkg/features/list.go` to `pkg/env/list.go`. This would result in the potential possibility of our end users enabling it accidentally, but may be this is not a real risk?

Reply:
Damn, that didn't work either. Is this possible that it does not work for e2e tests?

I even added logging in the function that handles this event:

```
"Received call to GetNodeVulnerabilities. RHCOSNodeScanning=false (false)"
```
while the tests claim:

```
$ go test -tags e2e -timeout=10s -count=1 -v -run ^TestGRPCGetRHCOSNodeVulnerabilities$  github.com/stackrox/scanner/e2etests
=== RUN   TestGRPCGetRHCOSNodeVulnerabilities
    node_scan_rhcos_test.go:108: FF set to: true
(...)
=== RUN   TestGRPCGetRHCOSNodeVulnerabilities/Uncertified_scan_is_unsupported_for_RHCOS_and_returns_no_features
--- FAIL: TestGRPCGetRHCOSNodeVulnerabilities (0.66s)
    --- FAIL: TestGRPCGetRHCOSNodeVulnerabilities/Selected_vulnerabilities_should_be_returned_by_the_certified_scan (0.47s)
    --- PASS: TestGRPCGetRHCOSNodeVulnerabilities/Uncertified_scan_is_unsupported_for_RHCOS_and_returns_no_features (0.19s)
FAIL
FAIL	github.com/stackrox/scanner/e2etests	1.347s
FAIL
```

Reply:
I will see what CI says, maybe this is an issue with the local running of those tests.

Reply:
Nope, it didn't help. I reverted the excessive logging, (rebased), and I am curious 🤔  whether we have any other ideas to resolve this.